File: backend/src/models/Export.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Export = sequelize.define('Export', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
    allowNull: false,
  },
  task_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  user_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  format: {
    type: DataTypes.STRING(10),
    allowNull: false,
    validate: {
      isIn: [['EXCEL', 'PDF']],
    },
  },
  filename: {
    type: DataTypes.STRING(255),
    allowNull: true,
  },
  record_count: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
  },
  file_path: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
  },
}, {
  tableName: 'exports',
  timestamps: false,
});

module.exports = Export;

File: backend/app/models/export.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Text, func
from sqlalchemy.orm import relationship
from app.database.database import Base


class Export(Base):
    __tablename__ = "exports"

    id = Column(Integer, primary_key=True, index=True)
    task_id = Column(Integer, ForeignKey("scraping_tasks.id"), nullable=False)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)

    format = Column(String(10), nullable=False)  # CSV/EXCEL
    filename = Column(String(255), nullable=True)
    record_count = Column(Integer, default=0)
    file_path = Column(Text, nullable=True)
    created_at = Column(DateTime, server_default=func.now())

    task = relationship("ScrapingTask", back_populates="exports")
    user = relationship("User", back_populates="exports")
